Tasks/google-play-rollout-update/google-play-rollout-update.ts

```diff
@@ -40,72 +46,67 @@ interface GlobalParams {
+        const packageName: string = tl.getPathInput('packageName', true);
+        const userFraction: number = Number(tl.getInput('userFraction', false)); // Used for staged rollouts
 
         // Constants
-        let GOOGLE_PLAY_SCOPES: string[] = ['https://www.googleapis.com/auth/androidpublisher'];
-        let globalParams: GlobalParams = { auth: null, params: {} };
-
-        let jwtClient: any = new google.auth.JWT(key.client_email, null, key.private_key, GOOGLE_PLAY_SCOPES, null);
-        let edits: any = publisher.edits;
+        const GOOGLE_PLAY_SCOPES: string[] = ['https://www.googleapis.com/auth/androidpublisher'];
+        const globalParams: GlobalParams = { auth: null, params: {} };
 
-        [edits, edits.tracks, jwtClient].forEach(Promise.promisifyAll);
+        const jwtClient = new google.auth.JWT(key.client_email, null, key.private_key, GOOGLE_PLAY_SCOPES, null);
+        const edits: any = publisher.edits;
 
         globalParams.auth = jwtClient;
         updateGlobalParams(globalParams, 'packageName', packageName);
 
         console.log(tl.loc('Authenticating'));
-        let currentEdit: any = jwtClient.authorizeAsync().then(function (res) {
-            return getNewEdit(edits, globalParams, packageName);
-        });
-
-        currentEdit = currentEdit.then(function (res) {
-            console.log(tl.loc('GetTrackRolloutInfo'));
-            return getTrack(edits, packageName, 'rollout');
-        });
-
-        currentEdit = currentEdit.then(function (res) {
-            console.log(tl.loc('CurrentUserFrac', res[0].userFraction));
-            return updateTrack(edits, packageName, 'rollout', res[0].versionCodes, userFraction);
-        });
-
-        currentEdit = currentEdit.then(function (res) {
-            return edits.commitAsync().then(function (res) {
-                console.log(tl.loc('RolloutFracUpdate'));
-                tl.setResult(tl.TaskResult.Succeeded, tl.loc('Success'));
-            });
-        }).catch(function (err) {
-            console.error(err);
-            tl.setResult(tl.TaskResult.Failed, tl.loc('Failure'));
-        });
+        await jwtClient.authorize();
+        await getNewEdit(edits, globalParams, packageName);
+
+        console.log(tl.loc('GetTrackRolloutInfo'));
+        const track: any = await getTrack(edits, packageName, rolloutTrack);
+        tl.debug('Track' + JSON.stringify(track.data));
+        const inProgressTrack = track.data.releases.find(x => x.status === 'inProgress');
+        if (!inProgressTrack) {
+            throw new Error(tl.loc('InProgressNotFound'));
+        }
+
+        console.log(tl.loc('CurrentUserFrac', inProgressTrack.userFraction));
+        const updatedTrack: any = await updateTrack(edits, packageName, rolloutTrack, inProgressTrack.versionCodes, userFraction);
+        tl.debug('Update Track' + JSON.stringify(updatedTrack.data));
+
+        console.log(tl.loc('RolloutFracUpdate'));
+        const commit: any = await edits.commit();
```

Comment:
Why replace commitAsync() with commit()? Is that a change in the V3 API?

Reply:
`commitAsync` is from `Promisify` (on the removed line ~80).  Because we're not using bluebird, we don't have that anymore and instead I'm using `await`.